inference-engine/src/vpu/graph_transformer/src/stages/reduce.cpp

```diff
@@ -127,41 +127,43 @@ class ReduceStage final : public StageNode {
 }  // namespace
 
 void FrontEnd::parseReduce(const Model& model, const ie::CNNLayerPtr& _layer, const DataVector& inputs, const DataVector& outputs) const {
-    auto layer = std::dynamic_pointer_cast<ie::ReduceLayer>(_layer);
-    VPU_THROW_UNLESS(layer != nullptr,
-                     "Layer {} of type {} is nullptr",
-                     layer->name, layer->type);
+    VPU_THROW_UNLESS(_layer != nullptr,
+                     "parseReduce expects valid CNNLayerPtr, actually got nullptr");
+    const auto reduce = std::dynamic_pointer_cast<ie::ReduceLayer>(_layer);
+    VPU_THROW_UNLESS(reduce != nullptr,
+                     "Could not perform dynamic cast from {} layer with name {} to ie::ReduceLayer",
+                     _layer->type, _layer->name);
     VPU_THROW_UNLESS(inputs.size() == 2,
                      "Layer {} of type {} expects {} inputs, but provided {}",
-                     layer->name, layer->type, 2, inputs.size());
+                     _layer->name, _layer->type, 2, inputs.size());
```

Comment:
Don't see a reason to use `_layer` -> use `reduce` since you already casted it. Or cast near the place you need information about derived class. The point here is to arrange code in a way it's has maximum *locality*: all information appear only when it is needed and disappear once it's unused - so we reduce entropy